### ISSUE TITLE
Show the 'thousands' string only if it is needed

### ISSUE THREAD

**Issue by abias** [PoetOS/moodle-mod_questionnaire#393]
Steps to reproduce:

* Create a new questionnaire
* Add a 'Numeric' question
* Within the question configuration, set " Max. digits allowed" to 2. All other question settings can be set to arbitrary values.
* View the questionnaire in preview mode

Actual result:
* The question says "Do not use thousands separators." above of the input field
![grafik](https://user-images.githubusercontent.com/1092118/161918361-a15827bd-2a0b-4d24-8415-43d7df621bcd.png)

Expected result:
* The question just displays the input field

Analysis:
* In https://github.com/PoetOS/moodle-mod_questionnaire/blob/main/templates/question_numeric.mustache#L47, the 'thousands' string is shown regardless of any condition.
* For large numbers, this note makes sense. For small number, especially if the user can't enter more than 3 digits according to the question configuration, this note should not be shown at all.

**Comment by mchurchward** [comment]
Good idea. This would be a relatively small change to flag when the message should be shown.

**Comment by mchurchward** [comment]
I wonder if there should also be a "maximum digits" message somewhere, when this is place?

**Comment by mchurchward** [comment]
Please review the changes in [GHI393](https://github.com/PoetOS/moodle-mod_questionnaire/tree/GHI393) branch.

**Comment by abias** [comment]
@mchurchward  - Thanks for these changes. I tested the branch GHI393 successfully, it does what I expected.

> I wonder if there should also be a "maximum digits" message somewhere, when this is place?

You did not implement anything for this, did you forget this or drop this idea by purpose?

**Comment by mchurchward** [comment]
I didn't put the maximum digits message in. You hadn't responded, or if you did, I missed it. Happy to add that though if you want it.

**Comment by abias** [comment]
Yes, I had responded, but not here. I responded in our Telegram discussion which we had in parallel.

Here's what I had written there:
"I think the code just sets the maxlength of the input field. This is not really obvious. So, yes, that would be wise."

Thinking about this some more, often the number of digits is already clear from the question context (for example: 'Enter your year of birth' or 'Enter your age'). So, if you add this note everytime a numeric field is displayed, we might run into a situation again where the form is cluttered with an unnecessary note.

Summing up, I would propose to merge the branch as it is and close this off :)

**Comment by abias** [comment]
Hi Mike,

sorry for my late reply.
Yes, you can surely release the new version now :)

**Comment by abias** [comment]
As the new version is published now, I think this issue can be closed.
